### ISSUE TITLE
Simulated hand ray does not show until you move the hand

### ISSUE THREAD

**Issue by luis-valverde-ms** [microsoft/MixedRealityToolkit-Unity#5029]
## To reproduce

1. Play any MRTK scene with the default configuration profiles.
2. Bring up the right simulated hand by pressing 'Y'.
3. Observe how no hand ray is displayed.
4. Move hand by moving the mouse while pressing space.
5. Observe how hand ray is displayed.

![SimulatedHandRay](https://user-images.githubusercontent.com/47451126/60026425-73e11300-9693-11e9-82ab-322683d90d12.gif)

## Expected behavior

The hand ray shows from the beginning.

## Your Setup (please complete the following information)

- Unity Version: 2018.4.1.f1
- MRTK Version: development post RC 2.1 (79044680e42ea7)

## Target Platform (please complete the following information)

- Editor

## Additional context

The ray pointer is missing the input changed event raised by the ` SimulatedArticulatedHand` to notify of the initial index finger pose. This is caused by `InputSystemGlobalListener.lateInitialize` being set to true and thus delaying the registration of the pointer as a global listener to a later point when the event has already been raised.

**Comment by Alexees** [comment]
I would like to add that the same thing happens in the portal with the emulated motion controllers, with the addition that the transform gizmos are also only loaded when you move at least once.

**Comment by julenka** [comment]
Thanks for the great analysis, Luis. You pretty much figured it all out. Wish all bugs were written like that haha. 

I actually hit this bug when writing my bounding box tests. I worked around it by setting `        private bool lateInitialize = false;` in InputSystemGlobalHandlerListener. Not sure why lateInitialize needs to be true. I'm still working on my bounding box tests, might take another day. If nobody else fixes this in an earlier PR I can check it in.

**Comment by julenka** [comment]
This also fixes a bug where the hand ray shows at the origin until you move it (which I thought was an open issue but I cannot find it...)

**Comment by julenka** [comment]
I'm assigning myself because I have a fix but this is POACHABLE -- I will fix it as part of my PR for bounding box tests which might go out for PR in about a day.

**Comment by luis-valverde-ms** [comment]
@julenka , thanks for taking over this. A few things to bear in mind:

1. After #4970 , global listeners distributed with MRTK inherit from `InputSystemGlobalHandlerListener` instead.  `InputSystemGlobalListener` is still around for backwards compatibility.
2. Changing the default value of `lateInitialize` will change the behavior of all global listeners, which may have unintended consequences. That being said , I rather break something and understand why it breaks than leaving something in just in case without really knowing why is needed.
3. I'd check this fix in as a separate PR. The more targeted a PR is the easier it is to understand, test and review it.

**Comment by julenka** [comment]
Yup, I agree, I moved this as a separate PR so it's easier to Review. See #5044. And I misspoke, I changed InputSystemGlobalHandlerListener

**Comment by wiwei** [comment]
Closing this out, as I think this one has been addressed.
